### PR TITLE
libint: add version 2.9.0

### PR DESCRIPTION
**Pull request by snehring**
The second stage build process is  cmake from 2.6.0 on. It looks like they're in the process of switching to cmake with the first stage too, so we may be able to make this an actual CMakePackage in the future.

### CODE REVIEW

**Comment by snehring** [comment]
not sure what to make of the build failures on power

**Comment by snehring** [comment]
@spackbot re-run pipeline

**Comment by snehring** [comment]
Ah, it's choking on unicode characters in some of the variable names

**Comment by snehring** [comment]
Reproduced on `gcc@9.4.0` on x86_64. I'll see what I can do about it.

File: var/spack/repos/builtin/packages/libint/package.py [hunk 1]
```diff
@@ -212,39 +211,48 @@ def build(self, spec, prefix):
...
+            mkdirp("build")
+            with working_dir("build"):
+                cmake(*cmake_args)
+                make()
 
     @when("@2.6.0:")
     def check(self):
         with working_dir(os.path.join(self.build_directory, "generated")):
-            make("check")
+            with working_dir("build"):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libint/package.py]: Can we avoid changing dirs 2 times in a row?

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by snehring** [comment]
[Limitation](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=67224) in gcc before version 10 I'm afraid.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Waiting for @dev-zero to chime in. Ping me in a while to merge, in case nothing happens.

**Pull request merged**